src/main/java/teammates/common/util/FieldValidator.java

```diff
@@ -578,15 +584,46 @@ public static String getInvalidityInfoForTimeForVisibilityStartAndResultsPublish
                 SESSION_VISIBLE_TIME_FIELD_NAME, RESULTS_VISIBLE_TIME_FIELD_NAME);
     }
 
+    /**
+     * Checks if the session end time is before all extended deadlines.
+     * @return Error string if any deadline in {@code deadlines} is before {@code sessionEnd}, an empty one otherwise.
+     */
+    public static String getInvalidityInfoForTimeForSessionEndAndExtendedDeadlines(
+            Instant sessionEnd, Map<String, Instant> deadlines) {
+        return deadlines.entrySet()
+                .stream()
+                .map(entry -> getInvalidityInfoForFirstTimeIsStrictlyBeforeSecondTime(sessionEnd, entry.getValue(),
+                        SESSION_END_TIME_FIELD_NAME, EXTENDED_DEADLINES_FIELD_NAME))
+                .filter(invalidityInfo -> !invalidityInfo.isEmpty())
+                .findFirst()
+                .orElse("");
+    }
+
     private static String getInvalidityInfoForFirstTimeIsBeforeSecondTime(
             Instant earlierTime, Instant laterTime, String earlierTimeFieldName, String laterTimeFieldName) {
+        return getInvalidityInfoForFirstTimeComparedToSecondTime(earlierTime, laterTime, earlierTimeFieldName,
+                laterTimeFieldName, (firstTime, secondTime) -> secondTime.isBefore(firstTime),
+                TIME_BEFORE_ERROR_MESSAGE);
+    }
+
+    private static String getInvalidityInfoForFirstTimeIsStrictlyBeforeSecondTime(
+            Instant earlierTime, Instant laterTime, String earlierTimeFieldName, String laterTimeFieldName) {
+        return getInvalidityInfoForFirstTimeComparedToSecondTime(earlierTime, laterTime, earlierTimeFieldName,
+                laterTimeFieldName,
+                (firstTime, secondTime) -> secondTime.isBefore(firstTime) || secondTime.equals(firstTime),
+                TIME_BEFORE_OR_EQUAL_ERROR_MESSAGE);
+    }
```

Comment:
>really tired atm

I think I did this while tired too lol. I think you're right. I will swap them.

Reply:
Wait no sorry. What you say is correct, `firstTime` being strictly before `secondTime` means `firstTime < secondTime`, not `firstTime <= secondTime`, but the method is also correct. The reason why it is confusing is because it checks if the times are invalid, not if it is valid. So basically if `firstTime < secondTime` is the check for validity, then `firstTime >= secondTime` is the one for invalidity.

Maybe you can confirm also if this is correct? I hope I am not confused about this again.

Reply:
Oh ok, I see it now. The `secondTime.isBefore(firstTime)` threw me off, I saw it as `firstTime.isBefore(secondTime)` yesterday, sorry about that.

Reply:
I think it's confusing regardless, so I'd consider it bad code. You know what, I think I'll swap the invalidity check for a validity check instead and check it is not valid instead of checking it is invalid. Hopefully, it will be less confusing for future readers that way.